docs/azure_authentication.md

```diff
@@ -0,0 +1,43 @@
+# Azure Operator authentication with Azure
```

Comment:
I think this is related: https://github.com/giantswarm/giantswarm/blob/master/ops-recipes/azure-service-principals.md

Maybe you want to link it somewhere in this doc.

Reply:
Done